src/context/AuthContext.tsx

```diff
@@ -23,31 +13,40 @@ type AuthContextType = {
 
 const AuthContext = createContext<AuthContextType>({} as AuthContextType);
 
-export function AuthProvider({ children }: { children: ReactNode }) {
+export function AuthProvider({ children } : { children: ReactNode }) {
   const [token, setToken] = useState<string | null>(null);
 
   useEffect(() => {
     // On mount, check if a token is in localStorage
     const storedToken = localStorage.getItem('token');
-    if (storedToken) {
-      setToken(storedToken);
+    if (storedToken && !isTokenExpired(storedToken)) {
+     setToken(storedToken);
     }
   }, []);
 
+  const isTokenExpired = (token: string): boolean => {
+    const decoded: any = jwtDecode(token);
+    const currentTime = Date.now() / 1000;
+    return decoded.exp < currentTime;
+  };
+
   /**
    * LOG IN
    */
   async function login(username: string, password: string) {
-    const newToken = await loginUser({ username, password });
-    localStorage.setItem('token', newToken);
-    setToken(newToken);
+    try {
+      const newToken = await loginUser({ username, password });
+      localStorage.setItem('token', newToken);
+      setToken(newToken);
+    } catch (error) {console.error('Login failed', error)};
```

Comment:
This is not what we want, we want to show the errors to the user by setting up the Errors, on the Login page, if you catch the errors here, then we won't get them in the login page.
Remove try and catch, we do want to trigger AxiosError.